src/main/java/com/codegym/restaurant/service/impl/FoodTypeServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.codegym.restaurant.service.impl;
+
+import com.codegym.restaurant.exception.EntityRestoreFailedException;
+import com.codegym.restaurant.exception.FoodTypeException;
+import com.codegym.restaurant.exception.ShiftNotFoundException;
+import com.codegym.restaurant.model.bussiness.Food;
+import com.codegym.restaurant.model.bussiness.FoodType;
+import com.codegym.restaurant.model.hr.Shift;
+import com.codegym.restaurant.repository.FoodRepository;
+import com.codegym.restaurant.repository.FoodTypeRepository;
+import com.codegym.restaurant.service.FoodTypeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+public class FoodTypeServiceImpl implements FoodTypeService {
+    @Autowired
+    private FoodTypeRepository foodTypeRepository;
+    @Autowired
+    private FoodRepository foodRepository;
+
+    @Override
+    public List<FoodType> getAll() {
+        return foodTypeRepository.findAllAvailable();
+    }
+
+    @Override
+    public List<FoodType> getAllDeleted() {
+        return foodTypeRepository.findAllDeleted();
+    }
+
+    @Override
+    public FoodType getById(Integer id) {
+        return foodTypeRepository.findAvailableById(id)
+                .orElseThrow(() -> new FoodTypeException("Loại món không tồn tại"));
+    }
+
+    @Override
+    public FoodType create(FoodType foodType) {
+        return foodTypeRepository.save(foodType);
+    }
+
+    @Override
+    public FoodType update(FoodType foodType) {
+        return foodTypeRepository.save(foodType);
+    }
+
+    @Override
+    public void delete(Integer integer) {
+        FoodType foodType = getById(integer);
+        foodType.setDeleted(true);
+        foodTypeRepository.save(foodType);
+    }
```

Comment:
chỗ ni có 2 trường hợp, 1 là béo xóa foodType thì cũng xóa luôn toàn bộ food trong loại món nớ, 2 là béo kiểm tra coi trong loại món nớ có món mô chưa xóa không, nếu có thì quăng lỗi, không cho xóa. chọn 1 trong 2 rồi làm nha